source/optional_features/custom_metadata/index.html.md.erb

```diff
@@ -0,0 +1,50 @@
+---
+title: Custom metadata
+weight: 135
+---
+
+# Custom metadata
+
+You can add custom metadata to a new payment. For example, you can add a reference number from your finance or accounting system, so you can reconcile the payment against bank transaction information.
+
+You can add metadata when you make an API call to create a new payment. You cannot add metadata to Direct Debit transactions or payment links.
+
+End users cannot see metadata while they're making a payment.
+
+## Add metadata to a payment
+
+Include a `metadata` object in the body of your [Create new payment](/making_payments/#making-payments) request. For example:
+
+```
+"metadata": {
+   "ledger_code": "AB100",
+   "an_internal_reference_number": 200,
+...
+}
+```
+
+The `metadata` object must contain between 1 and 10 parameters.
+
+Each parameter key must be between 1 and 30 characters long.
+
+Each parameter value must be both:
+
+- no longer than 50 characters
+- a string, number or boolean data type
+
+Your metadata will be included in the API response.
+
+You cannot add or change the metadata after you've created the payment request.
+
+Refer to the <a
+href="https://govukpay-api-browser.cloudapps.digital/#newpayment"
+target="blank">API browser</a> or the <a href="https://github.com/alphagov/pay-publicapi/blob/master/swagger/swagger.json" target="blank">Swagger file</a> for more information.
+
+## See a payment's metadata
+
+When you [make an API call to get information about a payment](/reporting/#reporting-via-the-gov-uk-pay-api), the `metadata` object will be included in the API response.
```

Comment:
I can't seem to find the link `/reporting/#reporting-via-the-gov-uk-pay-api`...

Reply:
I think we need to say that if the same metadata key is used more than once we will only store one occurrence of it and do not guarantee which one. It is strongly recommended that metadata keys are unique.

Reply:
Good point @danworth - would this work?
`Each parameter key:

- must be between 1 and 30 characters long
- should be unique - or only one of the duplicate keys will be stored`

Reply:
@oswaldquek That link seemed to work for me - it should link to https://docs.payments.service.gov.uk/reporting/#reporting-via-the-gov-uk-pay-api

Reply:
ah ok!

Reply:
yes that sounds good.